server/WebAPI/Controllers/WaypointFileController.cs

```diff
@@ -0,0 +1,86 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using WebAPI.Extension;
+using WebAPI.Services;
+using WebAPI.Services.Assets;
+
+namespace WebAPI.Controllers
+{
+  [Authorize]
+  [Route("api/[controller]")]
+  [ApiController]
+  public class WaypointFileController : ControllerBase
+  {
+
+    private readonly IWaypointFileService _waypointFileService;
+    private readonly IWaypointService _waypointService;
+
+    public WaypointFileController(IWaypointFileService waypointFileFileService, IWaypointService waypointService)
+    {
+      _waypointFileService = waypointFileFileService;
+      _waypointService = waypointService;
+    }
+
+    [HttpPost]
+    [Route("{id}")]
+    [Consumes("multipart/form-data")]
+    public async Task<IActionResult> UploadFile(int id, [FromForm] IFormFile file)
+    {
+      if (!_waypointService.IsWaypointAllowed(HttpContext.GetUserIdFromClaim(), id))
+        return Forbid();
+
+      if (file == null || file.Length < 0)
+        return BadRequest("fileIsEmpty");
+
+      if (file.Length > Consts.MaxWaypointFileSize)
+        return BadRequest("fileIsToBig");
+
+      if (!_waypointFileService.HasFreeSpace(id))
+        return BadRequest("NoSpaceAvailable");
+
+      return Ok(await _waypointFileService.AddFile(id, file));
+    }
+
+    [HttpPost]
+    [Route("multiple/{id}")]
+    [Consumes("multipart/form-data")]
+    public async Task<IActionResult> UploadFiles(int id, [FromForm] IFormFileCollection files)
```

Comment:
why use this id if the method doesn't respond to it? you need to add [FromRoute] or remove it from the router

Reply:
It works without explicit indication. If we want to add explicit argument indication, we have to add it into all controllers, even [FromBody]